packages/core/utils/lib/traverse/query-populate.js

```diff
@@ -27,10 +26,31 @@ const isStringArray = (value) => isArray(value) && value.every(isString);
 const populate = traverseFactory()
   // Array of strings ['foo', 'foo.bar'] => map(recurse), then filter out empty items
   .intercept(isStringArray, async (visitor, options, populate, { recurse }) => {
-    return Promise.all(populate.map((nestedPopulate) => recurse(visitor, options, nestedPopulate)));
+    const visitedPopulate = await Promise.all(
+      populate.map((nestedPopulate) => recurse(visitor, options, nestedPopulate))
+    );
+
+    return visitedPopulate.filter((item) => !isNil(item));
+  })
+  // Transform wildcard populate to an exhaustive list of attributes to populate.
+  // Avoid populating attributes from the database that will be sanitized in the output anyway.
```

Comment:
I think we can remove this comment as it only concern the content API :thinking: